### PR TITLE
gh-117084: Fix zip extraction bug

### PR DESCRIPTION
**Pull request by fliiiix**
On Windows a directory is ending in \\ and not /.

This is the smallest fix i could think of. 
Not sure if there are edge cases with files containing `\\` at the end.

* Issue: gh-117084

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2024-03-20-15-54-25.gh-issue-117084.uj1NbR.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix extraction of  ZIP files created on windows
```
Comment (timofurrer) [inline Misc/NEWS.d/next/Core and Builtins/2024-03-20-15-54-25.gh-issue-117084.uj1NbR.rst]:
```suggestion
Fix extraction of zip files created on Windows
```

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution @fliiiix. I have two requirements:

* It should not change the behavior on non-Windows platforms by default. The logic should match the logic of the extraction code.
* Tests are required.

Creating specific tests is something non-trivial task, so instead of guiding you to find the right solution (which I myself didn't know at first), it was easier for me to do it myself (#117129). Anyway, thank you. I hope for fruitful cooperation in the future.

**Comment by fliiiix** [comment]
> so instead of guiding you to find the right solution

Did you just denied my first contribution to cpython? :cry:

**Comment by serhiy-storchaka** [comment]
I'm sorry. But now that you've taken the first steps, cloned the repository, and created your first PR, creating other PRs should be much easier, so I look forward to your contribution in the future.

**Pull request closed without merging**